pkg/util/fsm/fsm.go

```diff
@@ -0,0 +1,122 @@
+package fsm
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	api "github.com/kubeedge/kubeedge/pkg/apis/fsm/v1alpha1"
+)
+
+type FSM struct {
```

Comment:
The business scope of this FSM seems to be for nodes, is this pkg `pkg/util/fsm` appropriate?

Reply:
This FSM is applicable not only to nodes, but also to the entire task.